framework/Metrics/SklMetric.py

```diff
@@ -65,18 +65,23 @@ def _localReadMoreXML(self,xmlNode):
   def distance(self, x, y=None, **kwargs):
     """
       This method returns the distance between two points x and y. If y is not provided then x is a pointSet and a distance matrix is returned
-      @ In, x, dict, dictionary containing data of x
-      @ In, y, dict, dictionary containing data of y
+      @ In, x, numpy.ndarray, array containing data of x
+      @ In, y, numpy.ndarray, array containing data of y
       @ Out, value, float or numpy.ndarray, distance between x and y (if y is provided) or a square distance matrix if y is None
     """
     if y is not None:
       if isinstance(x,np.ndarray) and isinstance(y,np.ndarray):
+        x = x if len(x.shape) != 1 else x.reshape(1,-1)
+        y = y if len(y.shape) != 1 else y.reshape(1,-1)
```

Comment:
Hm, possibly this should either print a warning, or reshaping arrays should be documented in the distance docstring.

Reply:
Good point, I will add a warning.